opensrp-app/src/main/java/org/smartregister/repository/TaskRepository.java

```diff
@@ -139,13 +159,15 @@ public void addOrUpdate(Task task, boolean updateOnly) {
             contentValues.put(STATUS, task.getStatus().name());
         }
         contentValues.put(BUSINESS_STATUS, task.getBusinessStatus());
-        contentValues.put(PRIORITY, task.getPriority());
+        contentValues.put(PRIORITY, task.getPriority().name());
```

Comment:
Is the `task.restriction` supposed to be saved here also?

Reply:
yes, added saving logic from line 168